phi2/phi2.py

```diff
@@ -202,7 +202,10 @@ def load_model():
 
     tokens = []
     for token, _ in zip(generate(prompt, model), range(args.max_tokens)):
-        tokens.append(token)
+        if token == tokenizer.eos_token_id:
+            break
+        else:
+            tokens.append(token)
 
         if (len(tokens) % 10) == 0:
             mx.eval(tokens)
```

Comment:
You should do the check inside the conditional otherwise you are evaling the model for every token.

Reply:
Maybe I am misunderstanding. I am not changing how often the eval happens. That seems like an unrelated change.

Reply:
No you are implicitly changing it :). When you do `token == tokenizer...` we `mlx` will implicitly evaluate the result of `token` under the hood in order to use it for the control flow in the program. If you did not have that line then `token` is not evaluated until line 211 when we explicitly call `mx.eval`.

Reply:
ah correct due to lazy eval